app/src/main/java/io/appwrite/almostnetflix/feed/MovieDataSource.kt

```diff
@@ -56,27 +59,30 @@ class MovieDataSource(
     }
 
     suspend fun getMyWatchlist(userId: String, movieIds: List<String>): List<Movie> {
-        val movies = database.listDocuments(
+        val movies = databases.listDocuments(
+            databaseId = Configuration.DATABASE_ID,
             collectionId = Configuration.MOVIE_COLLECTION_ID,
             queries = listOf(
-                Query.equal("\$id", movieIds)
+                Query.equal("\$id", value = movieIds),
```

Comment:
This causes the app to crash if `movieIds` is an empty array. @loganSims, does that happen to you?

Reply:
Think I found the problem. Apologies for missing this in my testing.

I had removed the `toString()` on the movieIds array, however looks like I can't just pass the array either.

Needed to change the whole queries value to something like the following:
```
    queries = movieIds.map {
        Query.equal("\$id", value = it)
    }         
```

But then we also need to check if the movieIds array is empty, otherwise the whole collection gets returned with a null query. So I can add something like this to the start of `getMyWatchlist`
```
    if (movieIds.isEmpty()) {
            return emptyList()
    }
```

Let me know if that sounds good to you and I can push up those changes for review. Or if there is a better way to handle the array being empty with a Query/etc. Still getting familiar with the Appwrite SDK so I might be missing something..

Reply:
```kotlin
    if (movieIds.isEmpty()) {
            return emptyList()
    }
```

makes sense.